### PR TITLE
Use liter version of repeaterjs

### PR DESCRIPTION
**Pull request by maneetgoyal**
Used a liter version of https://github.com/repeaterjs/repeater.

---

`pnpm i` is failing because of:

```
 WARN  Package name mismatch found while reading {"integrity":"sha1-AmNJ4SOUtpCatG/+uQxrx+xTkKg=","tarball":"https://registry.gadget.dev/npm/_/tarball/1268/1362/6391"} from the store. This means that the lockfile is broken. Expected package: @gadget-client/related-products-example@1.860.0-development.859. Actual package in the store by the given integrity: @gadget-client/related-products-example@1.859.0.
 ERR_PNPM_UNEXPECTED_PKG_CONTENT_IN_STORE  The lockfile is broken! A full installation will be performed in an attempt to fix it.
/Users/maneet/Documents/js-clients-1/packages/blog-example:
 ERR_PNPM_FETCH_404  GET https://registry.npmjs.org/@gadget-client%2Fblog: Not Found - 404

This error happened while installing a direct dependency of /Users/maneet/Documents/js-clients-1/packages/blog-example

@gadget-client/blog is not in the npm registry, or you have no permission to fetch it.
```

---

## PR Checklist

- [ ] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (maneetgoyal) [inline pnpm-lock.yaml:1]: @airhorns I tried downgrapding `pnpm` but this kept happening. Which `pnpm` version are you on?

Comment (maneetgoyal) [reply pnpm-lock.yaml:1]: I'm on `8.15.4`.

File: packages/api-client-core/package.json [hunk 2]
```diff
@@ -28,8 +28,9 @@
     "prerelease": "gitpkg publish"
   },
   "dependencies": {
+    "@manufac/repeater": "^4.0.6",
```
Comment (maneetgoyal) [inline packages/api-client-core/package.json:31]: @airhorns You have been added to this repo. I'll try to move all that code to this monorepo itself via subsequent PRs.

File: pnpm-lock.yaml (hunk 1 above)
Comment (airhorns) [reply pnpm-lock.yaml:1]: Weird. I updated `main` to lock us all to use the same version of pnpm via corepack. If you rebase on top of `main` and then re-run pnpm, I'll take whatever lockfile that produces!

**Comment by airhorns** [comment]
Ah, that PNPM store issue happened to me locally and is a result of a gadget bug. To fix it, run this:

```
rm -rf node_modules
pnpm store prune
pnpm install
```

and you should be back in business, it isn't actually the lock file that is busted but a bug in what version of a package tarball gadget served in the past.

**Pull request closed without merging**